stellar/its.js

```diff
@@ -23,6 +39,156 @@ async function removeTrustedChain(wallet, _, chain, contractConfig, arg, options
+}
+
+async function createPayGasAndCallContractAuth(spenderScVal, payload, gasTokenAddress, gasFeeAmount, chain, wallet) {
+    const { interchain_token_service: { address: itsAddress, initializeArgs: { gasServiceAddress, gatewayAddress, itsHubAddress } } = {} } =
+        chain.contracts;
+
+    const itsAddressScVal = nativeToScVal(Address.fromString(itsAddress), { type: 'address' });
+    const itsHubChainScVal = nativeToScVal(HUB_CHAIN, { type: 'string' });
+    const itsHubAddressScVal = nativeToScVal(itsHubAddress, { type: 'string' });
+    const gasServiceAddressScVal = nativeToScVal(gasServiceAddress, { type: 'address' });
+    const payloadScVal = nativeToScVal(Buffer.from(arrayify(payload, 'hex')), { type: 'bytes' });
+    const gasTokenScVal = tokenToScVal(gasTokenAddress, gasFeeAmount);
+    const gasFeeAmountScVal = nativeToScVal(gasFeeAmount, { type: 'i128' });
+    const emptyBytesScVal = nativeToScVal(Buffer.from(''), { type: 'bytes' });
+
+    const validUntil = await new rpc.Server(chain.rpc).getLatestLedger().then((info) => info.sequence + 100);
+
+    const transferAuth = new xdr.SorobanAuthorizedInvocation({
+        function: createAuthorizedFunc(Address.fromString(gasTokenAddress), 'transfer', [
+            spenderScVal,
+            gasServiceAddressScVal,
+            gasFeeAmountScVal,
+        ]),
+        subInvocations: [],
+    });
+
+    return Promise.all(
+        [
+            new xdr.SorobanAuthorizedInvocation({
+                function: createAuthorizedFunc(Address.fromString(gasServiceAddress), 'pay_gas', [
+                    itsAddressScVal,
+                    itsHubChainScVal,
+                    itsHubAddressScVal,
+                    payloadScVal,
+                    spenderScVal,
+                    gasTokenScVal,
+                    emptyBytesScVal,
+                ]),
+                subInvocations: [transferAuth],
+            }),
+            new xdr.SorobanAuthorizedInvocation({
+                function: createAuthorizedFunc(Address.fromString(gatewayAddress), 'call_contract', [
+                    itsAddressScVal,
+                    itsHubChainScVal,
+                    itsHubAddressScVal,
+                    emptyBytesScVal,
+                ]),
+                subInvocations: [],
+            }),
+        ].map((auth) => authorizeInvocation(wallet, validUntil, auth, wallet.publicKey(), getNetworkPassphrase(chain.networkType))),
+    );
+}
+
+async function deployRemoteCanonicalToken(wallet, _, chain, contractConfig, args, options) {
+    const itsAddress = contractConfig.address;
+    const spenderScVal = nativeToScVal(Address.fromString(wallet.publicKey()), { type: 'address' });
+
+    const [tokenAddress, tokenId, tokenName, tokenSymbol, decimals, destinationChain, gasTokenAddress, gasFeeAmount] = args;
+
+    const payload = defaultAbiCoder.encode(
+        ['uint8', 'string', 'bytes'],
+        [
+            3, // HubMessage type for SendToHub
```

Comment:
avoid magic constants, define constants under `common/its.js` so all scripts can use it